compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2468,7 +2481,7 @@ object Parsers {
         else if (isDefIntro(localModifierTokens))
           if (in.token == IMPLICIT || in.token == ERASED) {
             val start = in.offset
-            var imods = modifiers(funArgMods)
+            var imods = modifiers(funArgMods, parsedSeparately = BitSet(LAZY))
```

Comment:
I don't think we need this, the current error message is good:
```scala
72 |  lazy def bar = 1
   |           ^
   |           modifier `lazy` is not allowed for this definition
```

Reply:
That’s not the error message I’m fixing. Will separate the PR and add examples.